agent/consul/internal_endpoint.go

```diff
@@ -163,15 +172,31 @@ func (m *Internal) KeyringOperation(
 		switch args.Operation {
 		case structs.KeyringList:
 			if rule.KeyringRead(nil) != acl.Allow {
-				return fmt.Errorf("Reading keyring denied by ACLs")
+				_, ident, err := m.srv.ResolveIdentityFromToken(args.Token)
+				if err != nil {
+					m.srv.logger.Printf("[DEBUG] failed to fetch ACL AccessorID, err=%v", err)
+				}
+				var accessorID string
+				if ident != nil {
+					accessorID = ident.ID()
+				}
+				return fmt.Errorf("Reading keyring denied by ACLs, accessorID=%v", accessorID)
```

Comment:
Does this error get propagated back to the RPC requester? If so they already know their own token and we shouldn't need the accessor id here. Or do we log this elsewhere?

Reply:
Yeah, you're right that RPC requesters all have access to the secretID anyway, so we don't have to send it back w/ the error. Here's the usages I could find after looking for RPC callers of `Internal.KeyringOperation`

<img width="1007" alt="Screen Shot 2020-01-24 at 3 47 12 PM" src="https://user-images.githubusercontent.com/938395/73111889-f39cbd80-3ec0-11ea-90d7-395d0240e5c0.png">

Extracted the field from the `fmt.Errorf()` call here in latest revision